src/Compilers/VisualBasic/Portable/Analysis/FlowAnalysis/ReadWriteWalker.vb

```diff
@@ -143,6 +144,21 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
                     If Not (Me._currentMethodOrLambda = param.ContainingSymbol) Then
                         Me._captured.Add(param)
                     End If
+                Case SymbolKind.RangeVariable
+                    Dim range = DirectCast(variable, RangeVariableSymbol)
+                    If Me._currentMethodOrLambda = range.ContainingSymbol Then
+                        ' OK, defined in same method, so not captured
+                    ElseIf Me._currentQueryLambda Is Nothing Then
+                        ' Not in a query lambda, but defined and used in different methods, so captured.
+                        Me._captured.Add(range)
```

Comment:
Could you give an example of a scenario when we get here? Do we have a unit-test for it?